cmake/getversion.cmake

```diff
@@ -1,40 +1,215 @@
-find_package(Git)
-if(GIT_FOUND)
+# HELPER FUNCTIONS
+
+function(get_commit_id)
+	# get last commit hash
 	execute_process(
 		COMMAND ${GIT_EXECUTABLE} log -1 --abbrev=7 --date=short "--pretty=%h"
 		WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
 		RESULT_VARIABLE res_var
 		OUTPUT_VARIABLE GIT_COM_ID
 	)
-	if( NOT ${res_var} EQUAL 0 )
-		set( GIT_COMMIT_ID "unknown")
-		message( WARNING "Git failed (not a repo, or no tags). Build will not contain git revision info." )
+	if(NOT ${res_var} EQUAL 0)
+		message(WARNING "Git failed (not a repo, or no tags). Build will not contain git revision info.")
+		return()
 	endif()
-	string( REPLACE "\n" "" GIT_COMMIT_ID "${GIT_COM_ID}" )
 
+	string(REPLACE "\n" "" GIT_COM_ID "${GIT_COM_ID}")
+	set(GIT_COMMIT_ID "${GIT_COM_ID}" PARENT_SCOPE)
+	set(PROJECT_VERSION_LABEL "custom_${GIT_COM_ID}" PARENT_SCOPE)
+endfunction()
+
+function(get_commit_date)
+	# get last commit date
 	execute_process(
 		COMMAND ${GIT_EXECUTABLE} log -1 --date=short "--pretty=%cd"
 		WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
 		RESULT_VARIABLE res_var
 		OUTPUT_VARIABLE GIT_COM_DATE
 	)
-	if( NOT ${res_var} EQUAL 0 )
-		set( GIT_COMMIT_DATE "unknown")
-		set( GIT_COMMIT_DATE_FRIENDLY "unknown")
-		message( WARNING "Git failed (not a repo, or no tags). Build will not contain git revision info." )
+	if(NOT ${res_var} EQUAL 0)
+		message(WARNING "Git failed (not a repo, or no tags). Build will not contain git revision info.")
+		return()
+	endif()
+
+	string(REPLACE "\n" "" GIT_COM_DATE "${GIT_COM_DATE}")
+	set(GIT_COMMIT_DATE_FRIENDLY "${GIT_COM_DATE}" PARENT_SCOPE)
+
+	string(REPLACE "-" "" GIT_COM_DATE "${GIT_COM_DATE}")
+	set(GIT_COMMIT_DATE "${GIT_COM_DATE}" PARENT_SCOPE)
+endfunction()
+
+function(clean_release_name name)
```

Comment:
Sometimes people complain about `sbt` build definitions being scala code.

But that is so much better than this :(